src/pipeline/broad_phase/dbvt_broad_phase.rs

```diff
@@ -289,24 +280,24 @@ where
             }
         }
 
-        for (pair, _) in &mut self.pairs {
-            let proxy1 = self.proxies
-                .get(pair.0.uid())
-                .expect("DBVT broad phase: internal error.");
-            let proxy2 = self.proxies
-                .get(pair.1.uid())
-                .expect("DBVT broad phase: internal error.");
-
-            if proxy1.status == ProxyStatus::Deleted || proxy2.status == ProxyStatus::Deleted {
-                handler(&proxy1.data, &proxy2.data);
-                self.pairs_to_remove.push(*pair)
-            }
-        }
+        {
+            let proxies = &self.proxies;
+            self.pairs.retain(|pair, _| {
+                let proxy1 = proxies
+                    .get(pair.0.uid())
+                    .expect("DBVT broad phase: internal error.");
+                let proxy2 = proxies
+                    .get(pair.1.uid())
+                    .expect("DBVT broad phase: internal error.");
 
-        for pair in self.pairs_to_remove.iter() {
-            let _ = self.pairs.remove(pair);
+                if proxy1.status == ProxyStatus::Deleted || proxy2.status == ProxyStatus::Deleted {
+                    handler(&proxy1.data, &proxy2.data);
+                    true
+                } else {
+                    false
```

Comment:
This should be `false` at line 295 and `true` at line 297 since we want to retain the pair only if none of the proxies are `Deleted`.

Reply:
Indeed. Fixed it.